packages/blade/src/components/Input/BaseInput/StyledBaseInput.native.tsx

```diff
@@ -137,11 +137,25 @@ export const StyledBaseInput = React.forwardRef<
       numberOfLines,
       isTextArea,
       hasPopup,
+      shouldIgnoreBlurAnimation,
+      setShouldIgnoreBlurAnimation,
       ...props
     },
     ref,
   ) => {
     const buttonValue = props.value ? props.value : props.defaultValue;
+    const commonProps = {
+      onBlur: (): void => {
+        // In certain cases like SelectInput, we want to ignore the blur animation when option item is clicked.
+        // The selectinput should always look like it is in focus otherwise it triggers blur + focus again which can cause flicker
+        if (shouldIgnoreBlurAnimation && setShouldIgnoreBlurAnimation) {
```

Comment:
so if i pass `shouldIgnoreBlurAnimation={true}` then this will set it to false isn't weird? or am i not reading correctly?

Reply:
yup it will. It will ignore the first blur animation and set it to false.

We would only want to ignore the animation once right? otherwise the animation will never happen

Reply:
Actually you'll have to pass the state variable with it's set function. A bit hacky but alternative I can think of is to bring all the interaction logic outside BaseInput which won't make sense. 

We want to ignore the blur animation specifically when actionlistitem is clicked so that logic is outside of BaseInput. 

It's similar to how you would pass state control to child in React. You pass it's `value` and it's `setValue`.

Reply:
so wouldn't it be simpler if i just pass setter to the child components or just give the state to the child and then handle the setter internally? I'm just wondering that giving too many things might confuse the developer building on top of blade

Reply:
Can you check this now f76bfbf09676e6c6d1fb9f1066b37a23a9a6d35b

So handling the setters inside BaseInput was complex because setting is primarily dependent on the ActionListItem clicks. One way was to have `shouldIgnoreNextBlur` prop and then ignore that prop internally if it is already ignored once but that too included lots of complex conditions. Also things like what do we mean by "next" cause if it is always true then all events become next.

I am now handling it outside BaseInput in different state and passing the state to BaseInput.